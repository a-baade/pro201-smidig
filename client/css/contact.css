.contact-form{
    width: 100%;
    max-width: 650px;
    min-width: 500px;
    padding: 1em 4em 3em 4em;
    margin: 8px 0;
    box-sizing: border-box;
    border: 1px solid black;
    box-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    position: relative;
}

.contact-sidebar{
    width: 100%;
    max-width: 500px;
    min-width: 500px;
    background-color: #95bdc4;
    margin: 8px 0;
    padding: 2.2em 5.5em 0 5.5em;
    box-sizing: border-box;
    box-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 4px;
    position: relative;
    display: flex;
    flex-direction: column;
    text-align: center;
}

.contact-info{
    display: flex;
    flex-direction: row;
    justify-content: left;
    margin-top: 2em;
    position: relative;
    z-index: 2;
}

.sidebar-title{
    position: relative;
    z-index: 2;
}

.contact-icon{
    height: 1.4em;
    margin-top: 1.3vh;
    margin-right: 1vh;
}

.contact-input-field{
    margin-top: 2vh;
}

.contact-input{
    background-color: #e5e5e5;
}

.contact-textarea{
    height: 10em;
}

.submit-btn{
    margin-top: 1.5vh;
    font-weight: bold;
}

/* Background images */

.pink{
    position: absolute;
    margin-top: -24em;
    margin-left: -17em;
    opacity: 0.7;
}

.blue{
    width: 32em;
    position: absolute;
    margin-top: -31em;
    margin-left: -12em;
    z-index: 1;
    opacity: 0.25;
}

.person{
    position: absolute;
    width: 28em;
    margin-top: -0.6em;
    margin-left: -12em;
}