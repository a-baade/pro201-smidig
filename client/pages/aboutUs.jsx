import React from "react";
import MainBg from "../pics/about-us-pictures/main-bg.png";
import FirstLayerArt from "../pics/about-us-pictures/firstLayerArt.png";
import TopLayerArt from "../pics/about-us-pictures/topLayerArt.png";
import PovertyFirst from "../pics/about-us-pictures/povertyFirstLayer.png";
import PovertySecond from "../pics/about-us-pictures/povertytopLayer.png";
import ZeroHunger from "../pics/about-us-pictures/zeroHungerFirst.png";
import ZeroHungerTop from "../pics/about-us-pictures/seroHungerTop.png";
import GoodHealthFirst from "../pics/about-us-pictures/goodHealthFirst.png";
import GoodHealthTop from "../pics/about-us-pictures/goodHealthTop.png";
import QualityEducation from "../pics/about-us-pictures/qualityEducation.png";
import QualityEducationTop from "../pics/about-us-pictures/qualityEducationTop.png";
import GenderEquality from "../pics/about-us-pictures/genderEqualityFirst.png";
import GenderEqualityTop from "../pics/about-us-pictures/genderEqualityTop.png";

export function AboutUs() {
  return (
    <div>
      <div>
        <img src={MainBg} className={"main-bg-pic"} alt={"mainBg"}/>
      </div>
      <div className={"second-row-box"}>
        <div className={"knowledge-text-box"}>
          <p className={"our-mission"}>OUR MISSION</p>
          <h1 className={"knowledge-power"}>KNOWLEDGE = POWER</h1>
          <p>We're a Norwegian female-founded startup changing corporate giving and the way companies and nonprofits connect.</p>
          <p>We're saving companies resources, help create stable income streams for nonprofits, and create deeper relationships between donor and nonprofit.</p>
          <p> We help implement solid strategies for impactful and sustainable giving, all in line with the UN SDGs.</p>
        </div>
        <div className={"art-box"}>
          <img src={FirstLayerArt} className={"first-layer-art"} alt={"firstLayer"}/>
          <img src={TopLayerArt} className={"top-layer-art"} alt={"topLayer"}/>
          <p className={"our-mission mission"}>OUR MISSION</p>
          <h1 className={"action-call"}>A CALL TO ACTION</h1>
        </div>
      </div>
      <div className={"big-container"}>
        <div className={"small-box light"}>
          <p className={"small-title"}>No poverty</p>
          <p className={"small-txt"}>We are committed to end poverty in all its forms everywhere</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Zero hunger</p>
          <p className={"small-txt"}>End hunger, achieve food security and improve nutrition and promote sustainable agriculture</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <img src={GoodHealthFirst} className={"good-health"}/>
          <img src={GoodHealthTop} className={"good-health-top"}/>
          <p className={"small-title good-health-title"}>Good health</p>
          <p className={"small-txt good-health-txt"}>Ensure healthy lives and promote well-being for all at all ages</p>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Quality education</p>
          <p className={"small-txt"}>Ensure inclusive and equitable quality education and promote lifelong learning opportunities for all</p>
          <img src={QualityEducation} className={"quality-education"}/>
          <img src={QualityEducationTop} className={"quality-education-top"}/>
        </div>
        <div className={"small-box dark"}>
          <img src={GenderEquality} className={"gender-equality"}/>
          <img src={GenderEqualityTop} className={"gender-equality-top"}/>
          <p className={"small-title gender-eql-title"}>Gender equality</p>
          <p className={"small-txt gender-eql-txt"}>Achieve gender equality and empower all women and girls</p>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Clean water & sanitation</p>
          <p className={"small-txt"}>Ensure availability and sustainable management of water and sanitation for all</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Affordable & clean energy</p>
          <p className={"small-txt"}>Ensure access to affordable, reliable, sustainable and modern energy for all</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Decent work & economic growth</p>
          <p className={"small-txt"}>Promote sustained, inclusive and sustainable economic growth, full and productive employment and decent work for all</p>
          <img src={ZeroHunger} className={"decent-work"}/>
          <img src={ZeroHungerTop} className={"decent-work-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Industry, innovation & infrastructure</p>
          <p className={"small-txt"}>Build resilient infrastructure, promote inclusive and sustainable industrialization and foster innovation</p>
          <img src={PovertyFirst} className={"industry"}/>
          <img src={PovertySecond} className={"industry-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Reduce inequalities</p>
          <p className={"small-txt"}>Reduce inequality within and among countries</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Subsainable cities & communinites</p>
          <p className={"small-txt"}>Make cities and human settlements inclusive, safe, resilient and sustainable</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>responsible consumption & production</p>
          <p className={"small-txt"}>Ensure sustainable consumption and production patterns</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Climate action</p>
          <p className={"small-txt"}>Take urgent action to combat climate change and its impacts</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Life below water</p>
          <p className={"small-txt"}>Conserve and sustainably use the oceans, seas and marine resources for sustainable development</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Life on land</p>
          <p className={"small-txt"}>Protect, restore and promote sustainable use of terrestrial ecosystems, sustainably manage forests, combat desertification, and halt and reverse land degradation and halt biodiversity loss</p>
          <img src={PovertyFirst} className={"life-first"}/>
          <img src={PovertySecond} className={"life-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Peace, justice & strong institution</p>
          <p className={"small-txt"}>Promote peaceful and inclusive societies for sustainable development, provide access to justice for all and build effective, accountable and inclusive institutions at all levels</p>
          <img src={ZeroHunger} className={"peace-first"}/>
          <img src={ZeroHungerTop} className={"peace-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Partnerships for the goals</p>
          <p className={"small-txt"}>Strengthen the means of implementation and revitalize the global partnership for sustainable development</p>
          <img src={PovertyFirst} className={"partnership-first"}/>
          <img src={PovertySecond} className={"partnership-second"}/>
        </div>

        <div className={"small-box dark"}>
          <img src={ZeroHunger} className={"un-goals-first"}/>
          <img src={ZeroHungerTop} className={"un-goals-top"}/>
          <p className={"un-goals-title"}>THE UN SDG GOALS</p>
        </div>
      </div>
    </div>
  );
}