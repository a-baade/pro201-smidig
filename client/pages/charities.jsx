import React, { useContext, useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { ApiContext } from "../apiContext";
import { useLoading } from "../useLoading";
import { fetchJSON } from "../lib/fetchJSON";
import LoadingWheel from "../components/loadingSpinner";

function CharityCard({
  charity: { _id, name, description, bgImage, charityLogo },
}) {
  // Charity card with button to donate or charity page
  return (
    <div className={"charity-card custom-shadow"}>
      <div
        className={"charity-background"}
        style={{ backgroundImage: `url( data:image/png;base64${bgImage} )` }}
      >
        <img
          src={`data:image/png;base64${charityLogo}`}
          className={"charity-img custom-shadow"}
        />
      </div>

      <div className={"charity-card-content"}>
        <h3>{name}</h3>
        <p>{description}</p>
        <div className={"card-buttons"}>
          <Link to={{ pathname: "/charities/charity/id?id=" + _id }}>
            <button className="card-button custom-shadow-button">
              Read more
            </button>
          </Link>
          <Link to={{ pathname: "/charities/donate/id?id=" + _id }}>
            <button className="card-button custom-shadow-button">Donate</button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export function Charities() {
  const { allCharities } = useContext(ApiContext);
  const { category, setCategory } = useState("");
  const { categoryQuery, setCategoryQuery } = useState("");
  const [search, setSearch] = useState("");
  const [data, setData] = useState([]);
  const { loading, error } = useLoading(
    async () => await allCharities({ category }),
    [category]
  );

  useEffect(async () => {
    if (search === "") {
      await fetchJSON("/api/charities").then((jsonData) => {
        setData(jsonData);
      });
    } else {
      await fetchJSON(`/api/charities/search/?name=${search}`).then(
        (jsonData) => {
          setData(jsonData);
        }
      );
    }
  }, [search]);

  function handleSubmitQuery(e) {
    e.preventDefault();
    setCategory(categoryQuery);
  }

  async function handleSearch(event) {
    await setSearch(event);
  }

  if (loading) {
    return <LoadingWheel />;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  // Add category filter
  return (
    <div className={"page-margin"}>
      <div className="container-charities">
        <div className="search-charities">
          <input
            className="search-input-charities"
            placeholder="Search Charities"
            value={search}
            onChange={(e) => handleSearch(e.target.value)}
          />
          {search && (
            <div className={"charity-results"}>Results for {search}</div>
          )}
        </div>
        <div></div>

        {/*<form onSubmit={handleSubmitQuery}>*/}
        {/*    <button  id={"category-query"}*/}
        {/*             value={categoryQuery}*/}
        {/*             onChange={(e) => setCategoryQuery(e.target.value)}>Water</button>*/}
        {/*    <button  id={"category-query"}*/}
        {/*             value={categoryQuery}*/}
        {/*             onChange={(e) => setCategoryQuery(e.target.value)}>Knowledge</button>*/}
        {/*</form>*/}
      </div>
      <div>
        <ul className={"charity-cards-container"}>
          {data?.map((charity, index) => (
            <CharityCard key={index} charity={charity} />
          ))}
        </ul>
      </div>
    </div>
  );
}
