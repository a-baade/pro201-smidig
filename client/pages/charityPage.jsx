import React, { useContext } from "react";
import { Link, useSearchParams} from "react-router-dom";
import { ApiContext } from "../apiContext";
import { useLoading } from "../useLoading";
import Arrow from "../pics/icons/arrow.png";
import NewsImg1 from "../pics/bg-pictures/img_3.png";
import NewsImg2 from "../pics/bg-pictures/img_4.png";
import NewsImg3 from "../pics/bg-pictures/img_5.jpg";
import Facebook from "../pics/icons/facebook.png";
import Instagram from "../pics/icons/instagram.png";
import Snapchat from "../pics/icons/snapchat.png";
import Tiktok from "../pics/icons/tiktok.png";
import WWW from "../pics/icons/world-wide-web.png";
import LoadingWheel from "../components/loadingSpinner";

function Charity({charity:
    { _id, name, description, bgImage, charityLogo }
}) {

  // Charity page
  return (
    <div>
      <Link to={"/charities"}>
        <div className={"back-button custom-shadow-button"}>
          <img src={Arrow} />
        </div>
      </Link>
      <img
        src={`data:image/png;base64${bgImage}`}
        className={"charity-page-background"}
      />
      <div className={"charity-page-content"}>
        <div className={"charity-page-description"}>
          <div className={"charity-description-top"}>
            <img
              src={`data:image/png;base64${charityLogo}`}
              className={"charity-description-logo"}
            />
            <h2>{name}</h2>
          </div>
          <div className={"charity-description-text"}>
            <p>{description}</p>
          </div>
        </div>
        <div className={"charity-page-side"}>
          <Link to={{ pathname: "/charities/donate/id?id=" + _id }}>
            <button className={"charity-page-donate-button custom-shadow-button"}>Donate</button>
          </Link>
          <a
            href={"https://www.google.com/"}
            className={"charity-social-links"}
          >
            <img src={WWW} /> examplecharity.com
          </a>
          <a
            href={"https://www.facebook.com/"}
            className={"charity-social-links"}
          >
            <img src={Facebook} /> Example Charity
          </a>
          <a
            href={"https://www.instagram.com/"}
            className={"charity-social-links"}
          >
            <img src={Instagram} /> @examplecharity
          </a>
          <a
            href={"https://www.tiktok.com/"}
            className={"charity-social-links"}
          >
            <img src={Tiktok} /> @examplecharity
          </a>
          <a
            href={"https://www.snapchat.com/"}
            className={"charity-social-links"}
          >
            <img src={Snapchat} /> examplecharity
          </a>
        </div>
      </div>
      <div className={"newsfeed-container custom-shadow"}>
        <h1>NEWSFEED</h1>
        <div className={"newsfeed-items-container"}>
          <div className={"newsfeed-item custom-shadow"}>
            <img
                src={NewsImg1}
                className={"newsfeed-image"}
                alt={"Afghan children in school"}
            />
            <p>
              Afghan refugee children enrolling in norwegian nurseries 2 weeks
              after coming to norway. Prime minister hopes that these efforts
              will help stabilize the everyday life of the children and that
              they will integrate quicker.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
                src={NewsImg2}
                className={"newsfeed-image"}
                alt={"Deers in nature"}
            />
            <p>
              Hunting season has ended, yet people are still hunting animals
              illegally. Protecting nature is our responsibility, and make sure
              that the next generation can enjoy wilderness like us.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
                src={NewsImg3}
                className={"newsfeed-image"}
                alt={"Akeem in karate training"}
            />
            <p>
              Thanks to the support of charities, children like Akeem are able
              to partake in after school activities. Children like Akeem would
              not be able to afford ...
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default function CharityPage() {
  let [searchParams, setSearchParams] = useSearchParams();
  const id = searchParams.get("id");

  const {oneCharity} = useContext(ApiContext);
  const {loading, error, data} = useLoading(
    async () => await oneCharity(id),
    []
  );

  if (loading) {
    return <LoadingWheel/>;;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  return (
    <div className={"page-margin"}>
      <ul className={"charity-page-container"}>
        {data.map((charity, index) => (
          <Charity key={index} charity={charity} />
        ))}
      </ul>
    </div>
  );
}
