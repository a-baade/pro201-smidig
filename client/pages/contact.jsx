import React from "react";
import Message from "../pics/sign-up-pic/img_2.png";
import Phone from "../pics/icons/phone.png";
import Location from "../pics/icons/location.png";
import Pink from "../pics/contact-pics/pink-squiggle.png";
import Blue from "../pics/contact-pics/blue-squiggle.png";
import Person from "../pics/contact-pics/person-flowers.png";

export function Contact() {
    return (
        <div className={"login-container page-margin"}>
            <div className={"contact-sidebar"}>
                <h1 className={"sidebar-title"}>We'd love to <br/> hear from you</h1>
                <div>
                    <div className={"contact-info"}>
                        <img src={Phone} className={"contact-icon"} alt={"phoneIcon"}/>
                        <p>+47 999 99 999</p>
                    </div>
                    <div className={"contact-info"}>
                        <img src={Message} className={"contact-icon"} alt={"msgIcon"}/>
                        <p>meliora.impact@gmail.com</p>
                    </div>
                    <div className={"contact-info"}>
                        <img src={Location} className={"contact-icon"} alt={"locationIcon"}/>
                        <p>ch. groghsgate 12, 0186 Oslo Norway</p>
                    </div>
                </div>
                <div className={"contact-background"}>
                    <img src={Pink} className={"pink"} alt={"pinkSquig"}/>
                    <img src={Blue} className={"blue"} alt={"blueSquig"}/>
                    <img src={Person} className={"person"} alt={"personPic"}/>
                </div>
            </div>

            <div className={"contact-form"}>
                <h1>Get in touch with us</h1>
                <p>Fill out the form and our team will get back to you <br/> within 72 hours.</p>
                <div className={"name-field contact-input-field"}>
                    <label className={"login-txt contact-txt"}>Your name</label>
                    <input type= "text" placeholder={"Firstname Lastname"} className={"login-txt contact-input"}/>
                </div>
                <div className={"email-field contact-input-field"}>
                    <label className={"login-txt contact-txt"}>Email address</label>
                    <input type= "email" placeholder={"email@address.com"} className={"login-txt contact-input"}/>
                </div>
                <div className={"message-field contact-input-field"}>
                    <label className={"login-txt contact-txt"}>Your message</label>
                    <input type="textarea" placeholder={"Write your message"} className={"login-txt contact-textarea contact-input"}/>
                </div>
                <button className={"submit-btn sign-in-btn"}>Submit</button>
            </div>
        </div>
    );
}