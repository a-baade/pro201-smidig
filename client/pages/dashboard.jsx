import React from "react";
import { Chart2 } from "../components/charts";
import { Chart1 } from "../components/charts";


export function Dashboard({ user }) {
  return (
      <div className={"dashboard-page"}>
          <div className={"dashboard-sidebar"}>
              <div className={"dashboard-pfp-name"}>
                  <img src={user.google.picture} alt="Profile picture" className={"dashboard-pfp"}/>
                  <p>{user.google.name}</p>
              </div>
              <p>My donations</p>
              <p>My subscriptions</p>
          </div>
          <div className={"dashboard-main"}>
              <div className={"dashboard-top-row"}>
                <div className={"dashboard-info-box"}>
                    <p>23 <br/> donation</p>
                </div>
                  <div className={"dashboard-info-box"}>
                      <p>5 <br/> subscriptions</p>
                  </div>
                  <div className={"dashboard-info-box"}>
                      <p>750 000<br/> donated</p>
                  </div>
              </div>
              <div className={"dashboard-row-2"}>
                  <h2>Donation over time</h2>
                  <div className={"chart-wrap"}>
                      <div className={"chart2"}>
                          <Chart2/>
                      </div>
                      <div className={"chart2-sidebar"}>
                          <select name="months" id="months">
                              <option value="january">January</option>
                              <option value="february">February</option>
                              <option value="march">March</option>
                              <option value="april">April</option>
                          </select>
                          <div className={"chart2-circle"}>
                              <p>192 351 <br/> Avg per month</p>
                          </div>
                      </div>
                  </div>
              </div>
              <div className={"dashboard-row-3"}>
                  <h2>Donasjoner</h2>
                  <h4>sammenlignet med lignende bedrifter</h4>
                  <div className={"chart-wrap"}>
                      <div className={"chart1"}>
                          <Chart1/>
                      </div>
                      <div className={"chart1-sidebar"}>
                          <div className={"chart1-circle"}>
                              <p className={"percent-txt"}>25%</p>
                              <p className={"chart1-circle-txt"}>Bak gjennomsnittet</p>
                          </div>
                          <div className={"chart1-textbox"}>
                              <p>Basert p√• firma <br/> sin omsetning</p>
                          </div>
                      </div>
                  </div>
              </div>
          </div>
      </div>
  );
}