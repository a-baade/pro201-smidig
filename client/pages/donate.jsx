import { ApiContext } from "../apiContext";
import { useLoading } from "../useLoading";
import React, { useContext } from "react";
import { Link, useSearchParams } from "react-router-dom";
import PlaceholderBackground from "../pics/bg-pictures/charities-placeholder.jpg";
import PlaceholderLogo from "../pics/bg-pictures/charities-logo-placeholder.png";

function DonateModule({ charity: { name } }) {
  // Replace placeholders with actual charities images

  // Add form functionality?
  return (
    <div className={"donate-page"}>
      <div className={"donate-page-top"}>
        <img src={PlaceholderBackground} />
      </div>
      <div className={"donate-page-content"}>
        <img src={PlaceholderLogo} className={"donate-logo"} />
        <h2>Donate to {name}</h2>
        <p>Select amount (minimum $ 10.00)</p>
        <div className={"donate-form-amount"}>
          <button>$ 100.00</button>
          <button>$ 50.00</button>
          <button>$ 25.00</button>
          <input type={"text"} placeholder={"Other amount"} />
        </div>
        <div className={"donate-text-div"}>
          <p>How often would you like to donate</p>
          <div className={"information-icon"}>
            ?
            <div className={"information-box"}>
              Choose how often you would like to donate selected amount to this
              charity. You can change amount and frequency at any time.
            </div>
          </div>
        </div>

        <div className={"donate-form-frequency"}>
          <div>
            <input
              type={"radio"}
              id={"monthly"}
              name={"donation-frequency"}
              value={"monthly"}
            />
            <label htmlFor={"monthly"}>Monthly</label>
          </div>
          <div>
            <input
              type={"radio"}
              id={"quarterly"}
              name={"donation-frequency"}
              value={"quarterly"}
            />
            <label htmlFor={"quarterly"}>Quarterly</label>
          </div>
          <div>
            <input
              type={"radio"}
              id={"semiannually"}
              name={"donation-frequency"}
              value={"semiannually"}
            />
            <label htmlFor={"semiannually"}>Semiannually</label>
          </div>
          <div>
            <input
              type={"radio"}
              id={"annually"}
              name={"donation-frequency"}
              value={"annually"}
            />
            <label htmlFor={"annually"}>Annually</label>
          </div>
        </div>
        <p className={"donate-text-button"}>
          Donate $ XXXX to {name} every XX month(s)
        </p>
        <div>
          <button className={"donate-page-button"}>
            Donate with Debit or Credit Card
          </button>
        </div>
      </div>
    </div>
  );
}

export default function Donate() {
  let [searchParams, setSearchParams] = useSearchParams();
  const id = searchParams.get("id");

  const { donateCharity } = useContext(ApiContext);
  const { loading, error, data } = useLoading(
    async () => await donateCharity(id),
    []
  );

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  return (
    <div>
      <ul className={"donate-page-container"}>
        {data.map((charity, index) => (
          <DonateModule key={index} charity={charity} />
        ))}
      </ul>
    </div>
  );
}
