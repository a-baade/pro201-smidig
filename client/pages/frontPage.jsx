import React from "react";
import FrontBg from "../pics/bg-pictures/water-pic.png";
import Nature from "../pics/bg-pictures/img.png";
import Welfare from "../pics/bg-pictures/img_1.png";
import Education from "../pics/bg-pictures/img_2.png";
import NewsImg1 from "../pics/bg-pictures/img_3.png";
import NewsImg2 from "../pics/bg-pictures/img_4.png";
import NewsImg3 from "../pics/bg-pictures/img_5.jpg";
import { Link } from "react-router-dom";

export function FrontPage() {
  return (
    <div className={"page-margin"}>
      <div
        className={"front-image"}
        style={{ backgroundImage: `url(${FrontBg})` }}
      >
        <h1 className={"front-image-text custom-font-knowledge"}>
          Meliora Impact
        </h1>
        <Link to={"/signup"}>
          <button className={"front-image-button custom-shadow-button"}>
            Join us
          </button>
        </Link>
      </div>

      <div className={"about-us-div"}>
        <div className={"about-us-title"}>
          <h1>About us</h1>
        </div>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
          viverra, eros at luctus molestie, dolor risus convallis orci, ut
          tempus tellus nunc quis nulla. Maecenas libero lectus, pharetra ut
          magna in, rutrum luctus lorem. Nulla blandit augue nulla, sit amet
          porta dolor interdum eu. Phasellus suscipit porttitor urna, non
          vestibulum dui volutpat quis. Donec viverra lectus diam, sed
          sollicitudin quam pretium at. Curabitur nisl ligula, porta nec finibus
          eu, venenatis vitae quam. Curabitur facilisis magna condimentum
          maximus feugiat. Aliquam erat volutpat. Mauris posuere vel mi non
          commodo. Nunc dapibus libero ipsum, placerat posuere augue vulputate
          vitae. Aenean volutpat sollicitudin accumsan. Vestibulum at maximus
          purus, in hendrerit eros. Phasellus id tellus a eros volutpat rutrum
          et in ligula. Donec quis odio augue. Mauris pellentesque feugiat lacus
          eget ornare.
        </p>
        <Link to={"/about"}>
          <button className={"about-us-button custom-shadow-button"}>
            Read more
          </button>
        </Link>
      </div>
      <div className={"topic-container"}>
        <div className={"topic"}>
          <h4>Protecting nature</h4>
          <div className={"topic-image"}>
            <img src={Nature} alt={"Picture of nature"} />
          </div>
          <p>
            Nature is by design sustainable, it has supported itself for decades
            but human intervention across the past centuries has been weakening
            the very infrastructure it stands on. We always aim to protect and
            invest in what is already here so we as well it stands a chance for
            future growth.
          </p>
        </div>
        <div className={"topic"}>
          <h4>Welfare</h4>
          <div className={"topic-image"}>
            <img src={Welfare} alt={"Picture of poor children"} />
          </div>
          <p>
            The clean water crisis persist as a leading issue in several large
            parts of the world. It is a human right and necessity yet is
            consistently overlooked and lacking in support. Our aim is to uplift
            and highlight these core violations of human rights to foster more
            support for the communities at large.
          </p>
        </div>
        <div className={"topic"}>
          <h4>Education</h4>
          <div className={"topic-image"}>
            <img src={Education} alt={"Picture of child in school"} />
          </div>
          <p>
            Studies prove that education plays a large role in how well people
            in communities of all ranges fare supporting themselves and breaking
            cycles generational poverty and trauma. We strongly believe it is a
            right to have access to quality education and learning which is why
            it is part of our focus.
          </p>
        </div>
      </div>
      <div className={"newsfeed-container custom-shadow"}>
        <h1>NEWSFEED</h1>
        <div className={"newsfeed-items-container"}>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg1}
              className={"newsfeed-image"}
              alt={"Afghan children in school"}
            />
            <p>
              Afghan refugee children enrolling in norwegian nurseries 2 weeks
              after coming to norway. Prime minister hopes that these efforts
              will help stabilize the everyday life of the children and that
              they will integrate quicker.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg2}
              className={"newsfeed-image"}
              alt={"Deers in nature"}
            />
            <p>
              Hunting season has ended, yet people are still hunting animals
              illegally. Protecting nature is our responsibility, and make sure
              that the next generation can enjoy wilderness like us.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg3}
              className={"newsfeed-image"}
              alt={"Akeem in karate training"}
            />
            <p>
              Thanks to the support of charities, children like Akeem are able
              to partake in after school activities. Children like Akeem would
              not be able to afford ...
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
        </div>
      </div>
    </div>
  );
}
