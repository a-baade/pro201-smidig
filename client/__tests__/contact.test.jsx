import React, { createElement } from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import { Contact } from "../pages/contact";

describe("contact component", () => {
  it("should test div element", async function () {
    const divElement = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Contact />
      </MemoryRouter>,
      divElement
    );
    expect(divElement.innerHTML).toMatchSnapshot();
  });
});
