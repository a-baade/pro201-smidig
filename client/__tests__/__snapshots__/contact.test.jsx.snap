// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact component should test div element 1`] = `"<div class=\\"login-container page-margin\\"><div class=\\"contact-sidebar\\"><h1 class=\\"sidebar-title\\">We'd love to <br> hear from you</h1><div><div class=\\"contact-info\\"><img src=\\"[object Object]\\" class=\\"contact-icon\\" alt=\\"phoneIcon\\"><p>+47 999 99 999</p></div><div class=\\"contact-info\\"><img src=\\"[object Object]\\" class=\\"contact-icon\\" alt=\\"msgIcon\\"><p>meliora.impact@gmail.com</p></div><div class=\\"contact-info\\"><img src=\\"[object Object]\\" class=\\"contact-icon\\" alt=\\"locationIcon\\"><p>ch. groghsgate 12, 0186 Oslo Norway</p></div></div><div class=\\"contact-background\\"><img src=\\"[object Object]\\" class=\\"pink\\" alt=\\"pinkSquig\\"><img src=\\"[object Object]\\" class=\\"blue\\" alt=\\"blueSquig\\"><img src=\\"[object Object]\\" class=\\"person\\" alt=\\"personPic\\"></div></div><div class=\\"contact-form\\"><h1>Get in touch with us</h1><p>Fill out the form and our team will get back to you <br> within 72 hours.</p><div class=\\"name-field contact-input-field\\"><label class=\\"login-txt contact-txt\\">Your name</label><input type=\\"text\\" placeholder=\\"Firstname Lastname\\" class=\\"login-txt contact-input\\"></div><div class=\\"email-field contact-input-field\\"><label class=\\"login-txt contact-txt\\">Email address</label><input type=\\"email\\" placeholder=\\"email@address.com\\" class=\\"login-txt contact-input\\"></div><div class=\\"message-field contact-input-field\\"><label class=\\"login-txt contact-txt\\">Your message</label><input type=\\"textarea\\" placeholder=\\"Write your message\\" class=\\"login-txt contact-textarea contact-input\\"></div><button class=\\"submit-btn sign-in-btn\\">Submit</button></div></div>"`;
